react-app/src/components/Result.js

```diff
@@ -36,6 +36,45 @@ function Result(props) {
     return total;
   }
 
+  function getLabel() {
+    // attempt to get MRN and name with fhirpath
+    const patient = fhirpath.evaluate(props.bundle, "Bundle.descendants().resource.where(resourceType='Patient')")[0];
+    if (patient) {
+      const mrn = patient.id;
+      const name = patient.name[0].text;
+
+      let isMasked = fhirpath.evaluate(patient, 'Patient.identifier.extension.valueCode')[0];
+
+      if (isMasked === 'masked') {
+        isMasked = true;
+      } else {
+        isMasked = false;
+      }
+
+      // if both MRN and name -- return string w / both
+      if (typeof mrn === 'string' && mrn.length > 0 && typeof name === 'string' && name.length > 0 && !isMasked) {
+        const label = mrn.concat(': ').concat(name);
+        return <p className="accordion-result-label">{label}</p>;
+      }
+      if (typeof name === 'string' && name.length > 0) {
+        return <p className="accordion-result-label">{name}</p>;
+      }
+      if (typeof mrn === 'string' && mrn.length > 0) {
+        return <p className="accordion-result-label">{mrn}</p>;
+      }
+    }
+
+    let label = 'Patient';
+    label = label.concat(' ').concat(props.id.toString());
+    return <p className="accordion-result-label">{label}</p>;
+
+    // if either MRN or name -- return string / the available one
+
+    // if neither MRN nor name -- "Patient " + patient_resource_id
+
+    // if no patient resource ID -- return "Patient " + props.id
```

Comment:
Could we reorganize these comments to wherever they should be?

Reply:
I have moved these comments.